packages/main/src/TimePicker.js

```diff
@@ -0,0 +1,640 @@
+			horizontalAlign: PopoverHorizontalAlign.Left,
+			allowTargetOverlap: true,
+			stayOpenOnScroll: true,
+			afterClose: () => {
+				this._isPickerOpen = false;
+				this.closePicker();
+			},
+		};
+
+		this._hoursParameters = {
+			minHour: 0,
+			maxHour: 0,
+			isTwelveHoursFormat: false,
+		};
+
+		this._slidersDomRefs = [];
+
+		if (!this.formatPattern) {
+			this.formatPattern = LocaleData.getInstance(getLocale()).getTimePattern(this.getFormat().oFormatOptions.style);
+		}
+	}
+
+	onBeforeRendering() {
+		if (!this.value) {
+			this.value = this.getFormat().format(new Date());
+		}
+		this._initHoursFormatParameters();
+	}
+
+	_handleInputChange() {
+		let nextValue = this._getInput().getInputValue();
+		const isValid = this.isValid(nextValue);
+
+		if (isValid) {
+			nextValue = this.normalizeValue(nextValue);
+		}
+
+		this.setValue(nextValue);
+		this.fireEvent("change", { value: nextValue, valid: isValid });
+		this.fireEvent("value-changed", { value: nextValue, valid: isValid });
+	}
+
+	_handleInputLiveChange() {
+		const nextValue = this._getInput().getInputValue(),
+			isValid = this.isValid(nextValue);
+
+		this.value = nextValue;
+		this.setSlidersValue();
+		this.fireEvent("input", { value: nextValue, valid: isValid });
+	}
+
+	setSlidersValue() {
+		const currentDate = this._getInput() ? this.getFormat().parse(this._getInput().getAttribute("value")) : null,
+			secondsSlider = this.getStaticAreaItemDomRef().querySelector(".ui5-timepicker-seconds-wheelslider"),
+			minutesSlider = this.getStaticAreaItemDomRef().querySelector(".ui5-timepicker-minutes-wheelslider"),
+			hoursSlider = this.getStaticAreaItemDomRef().querySelector(".ui5-timepicker-hours-wheelslider"),
+			periodsSlider = this.getStaticAreaItemDomRef().querySelector(".ui5-timepicker-period-wheelslider");
+
+		if (currentDate) {
+			if (hoursSlider) {
+				if (this._hoursParameters.isTwelveHoursFormat && currentDate.getHours() > this._hoursParameters.maxHour) {
+					hoursSlider.value = currentDate.getHours() - 12;
+				} else if (this._hoursParameters.isTwelveHoursFormat && currentDate.getHours() < this._hoursParameters.minHour) {
+					hoursSlider.value = currentDate.getHours() + 12;
+				} else {
+					hoursSlider.value = currentDate.getHours();
+				}
+			}
+			if (minutesSlider) {
+				minutesSlider.value = currentDate.getMinutes();
+			}
+			if (secondsSlider) {
+				secondsSlider.value = currentDate.getSeconds();
+			}
+			if (this._hoursParameters.isTwelveHoursFormat && periodsSlider && this._hoursParameters.minHour === 1) {
+				periodsSlider.value = currentDate.getHours() > this._hoursParameters.maxHour ? this.periodsArray[1] : this.periodsArray[0];
+			} else if (this._hoursParameters.isTwelveHoursFormat && periodsSlider) {
+				periodsSlider.value = (currentDate.getHours() > this._hoursParameters.maxHour || currentDate.getHours() === this._hoursParameters.minHour) ? this.periodsArray[1] : this.periodsArray[0];
```

Comment:
it is really hard to read this, please have a look and split the method into pieces